R/validation.R

```diff
@@ -107,3 +107,28 @@ validate_survey_and_foi_consistency_age_time <- function( #nolint
       "not exceed max age in survey_features."
       )
 }
+
+validate_foi_index <- function(
```

Comment:
I'd suggest checking that the FOI chunks are consecutive: e.g. that we don't get 1, 1, 1, 2, 2, 2, 1, 1, 3, 3. I think we also need to have indices for all integers up until the max. So we would raise a warning for 1, 1, 1, 3, 3 for instance because it misses a 2.